src/compiler/checker.ts

```diff
@@ -28820,6 +28833,9 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                 Debug.assert(parent.parent.kind === SyntaxKind.TemplateExpression);
                 return getContextualTypeForSubstitutionExpression(parent.parent as TemplateExpression, node);
             case SyntaxKind.ParenthesizedExpression: {
+                if (isJSDocSatisfiesExpression(parent)) {
+                    return getTypeFromTypeNode(getJSDocSatisfiesExpressionType(parent));
+                }
                 // Like in `checkParenthesizedExpression`, an `/** @type {xyz} */` comment before a parenthesized expression acts as a type cast.
                 const tag = isInJSFile(parent) ? getJSDocTypeTag(parent) : undefined;

```

Comment:
I would rather this logic all get gated on `isInJSFile`

Reply:
https://github.com/microsoft/TypeScript/blob/b25fc687b6135691f017998c01709185320301e4/src/compiler/utilities.ts#L9490

Reply:
I get that, I'd just rather do 1 check instead of 2 for the TS file case. Maybe it's fine this way.

Reply:
I've wrapped this logic into isInJSFile